subsys/bluetooth/host/gatt.c

```diff
@@ -549,10 +553,58 @@ enum delayed_store_flags {
 };
 
 #if defined(CONFIG_BT_SETTINGS_DELAYED_STORE)
-static void gatt_delayed_store_enqueue(struct bt_conn *conn, enum delayed_store_flags flag);
+static void gatt_delayed_store_enqueue(uint8_t id, const bt_addr_le_t *peer_addr,
+				       enum delayed_store_flags flag);
 #endif
 
 #if defined(CONFIG_BT_GATT_CACHING)
+bool set_change_aware_no_store(struct gatt_cf_cfg *cfg, bool aware)
+{
+	bool changed;
+
+	if (aware) {
+		changed = !atomic_test_and_set_bit(cfg->flags, CF_CHANGE_AWARE);
+	} else {
+		changed = atomic_test_and_clear_bit(cfg->flags, CF_CHANGE_AWARE);
+	}
+
+	LOG_DBG("peer is now change-%saware", aware ? "" : "un");
+
+	return changed;
+}
+
+void set_change_aware(struct gatt_cf_cfg *cfg, bool aware)
+{
+	bool changed = set_change_aware_no_store(cfg, aware);
+
+#if defined(CONFIG_BT_SETTINGS_DELAYED_STORE)
+	if (changed) {
+		gatt_delayed_store_enqueue(cfg->id, &cfg->peer, DELAYED_STORE_CF);
```

Comment:
With the current approach we may need to increase number of delayed CF writes that can be queued (to `(CONFIG_BT_MAX_PAIRED + CONFIG_BT_MAX_CONN)`). Keep in mind that we may call `set_change_aware` also for bonded peers that are not connected. Alternatively maybe we could simply store the value instantly if peer is not connected?

Reply:
I'd prefer the second solution, but it seems like an optimization. I'd rather do it after getting the bug fix in.

Reply:
The count seems updated, so it should work fine :)